src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -1181,10 +1186,12 @@ void RaftReplDev::on_log_found(logstore_seq_num_t lsn, log_buffer buf, void* ctx
     }
 
     rreq->set_lsn(repl_lsn);
+    rreq->set_lentry(lentry);
     rreq->init(rkey, jentry->code, false /* is_proposer */, entry_to_hdr(jentry), entry_to_key(jentry), data_size);
     RD_LOGD("Replay log on restart, rreq=[{}]", rreq->to_string());
 
     if (repl_lsn > m_rd_sb->durable_commit_lsn) {
+        commit_blk(rreq);
```

Comment:
Nit: can we dirty the allocator with these blks and still commit them in handle_commit()?. Just in the case for rollback.

Reply:
Is there a way to dirty the blks without allocating them?

Reply:
not yet but we can change dataservice api?

Reply:
this rreq will be handled in `handle_commit` line 1205, where `commit_blk` will be called.
I don`t think we need another `commit_blk` here

Reply:
also implement roll_back to free blk

Reply:
> Nit: can we dirty the allocator with these blks and still commit them in handle_commit()?. Just in the case for rollback.

Just be noted commit_blk will be expecting blks in allocated state for non-recovery mode (existing handle_commit), and for recovery mode, it will try to reserve on cache also before committing to disk; Need a bit more handling to combine them.